pkg/kubelet/kubelet.go

```diff
@@ -1918,7 +1907,7 @@ func (kl *Kubelet) setNodeStatus(node *api.Node) error {
 		if node.Status.Conditions[i].Type == api.NodeReady {
 			newCondition.LastTransitionTime = node.Status.Conditions[i].LastTransitionTime
 			if node.Status.Conditions[i].Status != api.ConditionTrue {
-				kl.recordNodeOnlineEvent()
+				kl.recordNodeStatusEvent("NodeReady")
```

Comment:
I must be missing something, but I don't understand the original logic (your change doesn't change the logic, just the way the event is recorded, but I want to make sure the logic is right). From the previous block of code in this function, it seems that the newCondition may be NodeReady==True or NodeReady==False. Yet the logic here seems to say "if the old condition was NodeReady!=True, then record NodeReady" even if the new condition is also NodeReady!=True. It also seems to not be updating LastTransitionTime, which seems wrong in the case where the new condition != the old condition.

The next line you changed (in the !updated section) also seems to be wrong for similar reasons.

The logic used for Unschedulable seems correct and is what I would have expected for NodeReady.

Can you explain what I'm missing?